eth/eventhandler/handlers.go

```diff
@@ -233,11 +233,21 @@ func (eh *EventHandler) handleShareCreation(
 		sharePublicKeys,
 		encryptedKeys,
 	)
+
+	var malformedEventError *MalformedEventError = nil
+
 	if err != nil {
-		return nil, fmt.Errorf("could not extract validator share from event: %w", err)
+		if errors.As(err, &malformedEventError) &&
+			malformedEventError.IsInvalidEncryptedShare &&
+			share.BelongsToOperator(eh.operatorData.GetOperatorData().ID) {
+
+			share.Metadata.Invalid = true
+		} else {
+			return nil, fmt.Errorf("could not extract validator share from event: %w", err)
+		}
```

Comment:
lets invert the condition to return if it's not the error we want, otherwise we continue with setting the invalid flag.